client/app/email-domain/email/responder/create/email-domain-email-responder-create.controller.js

```diff
@@ -66,12 +66,35 @@ angular.module("App").controller(
             this.model.responderDateEnd = "";
         }
 
+        resetResponderAccount () {
+            this.model.account = null;
+        }
+
         responderAccountCheck (input) {
-            input.$setValidity("email", this.model.account && validator.isEmail(`${this.model.account}@${this.domain}`));
+            input.$setValidity("email", _.isString(this.model.account) && validator.isEmail(`${this.model.account}@${this.domain}`));
             input.$setValidity("responder", _.indexOf(this.responders, this.model.account) === -1);
             input.$setValidity("account", _.indexOf(this.accounts, this.model.account) === -1);
         }
 
+        responderDatesCheck (start, end) {
+            this.responderDateStartCheck(start);
+            this.responderDateEndCheck(end);
+        }
+
+        responderDateStartCheck (input) {
+            if (!input.$dirty && !_.isEmpty(this.model.responderDateStart)) {
+                input.$setDirty();
+            }
+            input.$setValidity("date", !!this.model.responderDateStart && (!this.model.responderDateEnd || moment(this.model.responderDateStart).isBefore(this.model.responderDateEnd)));
+        }
+
+        responderDateEndCheck (input) {
+            if (!input.$dirty && !_.isEmpty(this.model.responderDateEnd)) {
+                input.$setDirty();
+            }
+            input.$setValidity("date", !!this.model.responderDateEnd && (!this.model.responderDateStart || moment(this.model.responderDateEnd).isAfter(this.model.responderDateStart)) && moment(this.model.responderDateEnd).isAfter(new Date()));
```

Comment:
`_.isString(this.model.responderDateEnd)` instead of `!!this.model.responderDateEnd` and the same goes for `!this.model.responderDateStart`

Actually the whole condition should be extracted for readability purposes